packages/jmix-react-antd/src/crud/editor/ui-callbacks/useEntityEditorCallbacks.ts

```diff
@@ -0,0 +1,56 @@
+import {NotificationType, notifications} from "../../../ui/notifications";
+import {useIntl} from "react-intl";
+import {useCallback, useMemo} from 'react';
+import {EntityEditorCallbacks, useParentScreen} from "@haulmont/jmix-react-web";
+import { EntityInstance } from "@haulmont/jmix-react-core";
+
+export function useEntityEditorCallbacks<TEntity, TData>(options: {
+  onCommit?: (entityInstance: EntityInstance<TEntity>) => void,
+  routingPath: string,
+}): EntityEditorCallbacks<TEntity, TData> {
+  const intl = useIntl();
+  const parentScreen = useParentScreen(options.routingPath);
+
+  const onCommit = useMemo(() =>
+    options.onCommit
+      ? (entityInstance: EntityInstance<TEntity>) => {
+        options.onCommit?.(entityInstance)
+        parentScreen();
```

Comment:
We should separate `onCommit` from UI updates.

I.e. when creating an entity, we will executre `onCommit` and the `onCreate` (we should probably have better names to underline which callback is responsible for UI and which for data).